### PR TITLE
Tests: Cookie-sync refine core logic of /cookie-sync endpoint 

### PR DESCRIPTION
**Pull request by marki1an**
Tests for https://github.com/prebid/prebid-server-java/pull/2065

### CODE REVIEW

File: src/test/groovy/org/prebid/server/functional/tests/CookieSyncSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,516 @@
...
+
+    PrebidServerService prebidServerService = pbsServiceFactory.getService(
+            ["adapters.generic.usersync.redirect.url"            : "$networkServiceContainer.rootUri/generic-usersync&redir={{redirect_url}}".toString(),
+             "adapters.generic.usersync.redirect.support-cors"   : "false",
+             "adapters.generic.usersync.redirect.format-override": "blank"])
+
+    def "PBS cookie sync with host-cookie.family which hasn't the same value in both uids and from the direct host-cookie then should re-sync"() {
+        given: "PBS bidders config"
+        def prebidServerService = pbsServiceFactory.getService(
+                ["host-cookie.family"                                : "generic",
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/CookieSyncSpec.groovy:38]: This name should correspond to bidder name. Please or should correlate with family name override. Please make this obvious.

File: src/test/groovy/org/prebid/server/functional/tests/CookieSyncSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,516 @@
...
+                 "host-cookie.cookie-name"                           : "rubicon",
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/CookieSyncSpec.groovy:39]: Please extract this value to a variable. And preferably make it random to signify that name can be arbitrary.

File: src/test/groovy/org/prebid/server/functional/tests/CookieSyncSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,516 @@
...
+                 "adapters.generic.usersync.redirect.url"            : "$networkServiceContainer.rootUri/generic-usersync&redir={{redirect_url}}".toString(),
+                 "adapters.generic.usersync.redirect.support-cors"   : "false",
+                 "adapters.generic.usersync.redirect.format-override": "blank"])
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/CookieSyncSpec.groovy:42]: Why do you need this here?

File: src/test/groovy/org/prebid/server/functional/tests/CookieSyncSpec.groovy [hunk 4]
```diff
@@ -0,0 +1,516 @@
...
+
+        and: "Default cookie sync request"
+        def cookieSyncRequest = CookieSyncRequest.defaultCookieSyncRequest
+
+        and: "Default uids cookie with rubicon"
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/CookieSyncSpec.groovy:47]: What do you mean by "with rubicon"? You're talking about host cookie here.

File: src/test/groovy/org/prebid/server/functional/tests/CookieSyncSpec.groovy [hunk 5]
```diff
@@ -0,0 +1,516 @@
...
+        def rubiconUidsCookie = UidsCookie.getDefaultUidsCookie(RUBICON)
+        def cookies = ["uids": UidsCookie.defaultUidsCookie, "rubicon": rubiconUidsCookie]
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/CookieSyncSpec.groovy:49]: This it totally wrong as host cookie is a plain string. And again, don't hardcode names.

File: src/test/groovy/org/prebid/server/functional/tests/CookieSyncSpec.groovy [hunk 6]
```diff
@@ -0,0 +1,516 @@
...
+
+        when: "PBS processes cookie sync request"
+        def response = prebidServerService.sendCookieSyncRequest(cookieSyncRequest, cookies)
+
+        then: "Response shouldn't contain error"
+        def bidderStatus = response.getBidderUserSync(GENERIC)
+        assert !bidderStatus.error
+
+        and: "Response should contain rubicon uids"
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/CookieSyncSpec.groovy:58]: Again, what do you mean by "rubicon uids"? You're once more talking about host cookie. But it's not clear to me what you are expecting.

File: src/test/groovy/org/prebid/server/functional/tests/CookieSyncSpec.groovy [hunk 7]
```diff
@@ -0,0 +1,516 @@
...
+        assert HttpUtil.decodeUrl(bidderStatus.userSync?.url)
+                       .contains("uid=${Base64.urlEncoder.encodeToString(encode(rubiconUidsCookie).bytes)}")
+    }
+
+    def "PBS cookie sync with host-cookie.family which hasn't the same value in both uids and from the direct host-cookie then should re-sync"() {
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/CookieSyncSpec.groovy:63]: Again, I can't understand what this test should check. This applies to most test names. They are not clear and have typos or state weird stuff.

**Comment by Net-burst** [comment]
Your branch also tracks an old state of feature branch. Please update.

**Review by Net-burst — CHANGES_REQUESTED** [review]

**Comment by Net-burst** [comment]
Your tests are arranged in a weird way. They should go in the order of increasing complexity of the feature. Secondary features should also go after primary.

File: src/test/groovy/org/prebid/server/functional/model/request/cookiesync/CookieSyncRequest.groovy [hunk 8]
```diff
@@ -18,13 +18,17 @@ class CookieSyncRequest {
     String usPrivacy
     @JsonProperty("coopSync")
     Boolean coopSync
+    Boolean debug
     Integer limit
     String account
+    @JsonProperty("filterSettings")
```
[the remark this answers is not in the document]
Comment (marki1an) [reply src/test/groovy/org/prebid/server/functional/model/request/cookiesync/CookieSyncRequest.groovy:24]: `@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy)` defined here

[the remark this answers is not in the document]
Comment (Net-burst) [reply src/test/groovy/org/prebid/server/functional/model/request/cookiesync/CookieSyncRequest.groovy:24]: This is precisely why I asked to remove this annotation.

[the remark this answers is not in the document]
Comment (Net-burst) [reply src/test/groovy/org/prebid/server/functional/model/request/cookiesync/CookieSyncRequest.groovy:24]: Sorry, my bad. I was sure that I'm seeing camelCase there...

**Review by Net-burst — APPROVED** [review]
LGTM

**Pull request merged**